analytic_engine/src/sst/parquet/hybrid.rs

```diff
@@ -0,0 +1,441 @@
+            type_str: self.datum_kind.to_string(),
+        })?;
+        let values_num = self.list_of_arrays.iter().map(|handle| handle.len()).sum();
+        let mut values = MutableBuffer::new(values_num * data_type_size);
+        let mut null_buffer = MutableBuffer::new_null(values_num);
+        let null_slice = null_buffer.as_slice_mut();
+
+        let mut length_so_far: i32 = 0;
+        offsets.push(length_so_far);
+        for array_handle in &self.list_of_arrays {
+            let shared_buffer = array_handle.data_slice();
+            let null_bitmap = array_handle.null_bitmap();
+
+            for (offset, length) in &array_handle.positions {
+                if let Some(bitmap) = null_bitmap {
+                    for i in 0..*length {
+                        if bitmap.is_set(i + *offset) {
+                            bit_util::set_bit(null_slice, length_so_far as usize + i);
+                        }
+                    }
+                } else {
+                    for i in 0..*length {
+                        bit_util::set_bit(null_slice, length_so_far as usize + i);
+                    }
+                }
+                length_so_far += *length as i32;
+                values.extend_from_slice(
+                    &shared_buffer[offset * data_type_size..(offset + length) * data_type_size],
+                );
+            }
+            offsets.push(length_so_far);
+        }
+        debug!(
+            "build_child_data offsets:{:?}, values:{:?}",
+            offsets.as_slice(),
+            values.as_slice()
+        );
+
+        let values_array_data = ArrayData::builder(self.datum_kind.to_arrow_data_type())
+            .len(values_num)
+            .add_buffer(values.into())
+            .null_bit_buffer(Some(null_buffer.into()))
+            .build()
+            .map_err(|e| Box::new(e) as _)
+            .context(EncodeRecordBatch)?;
+
+        Ok(values_array_data)
+    }
+
+    /// This function is an translation of [GenericListArray.from_iter_primitive](https://docs.rs/arrow/20.0.0/src/arrow/array/array_list.rs.html#151)
+    fn build(self) -> Result<ListArray> {
+        assert!(!self.list_of_arrays.is_empty());
+
+        let array_len = self.list_of_arrays.len();
+        let mut offsets =
+            MutableBuffer::new(self.list_of_arrays.len() * std::mem::size_of::<i32>());
+        let child_data = self.build_child_data(&mut offsets)?;
+        let field = Box::new(Field::new(
+            LIST_ITEM_NAME,
+            self.datum_kind.to_arrow_data_type(),
+            true,
+        ));
+        let array_data = ArrayData::builder(DataType::List(field))
+            .len(array_len)
+            .add_buffer(offsets.into())
+            .add_child_data(child_data);
+
+        // TODO: change to unsafe version?
+        // https://docs.rs/arrow/20.0.0/src/arrow/array/array_list.rs.html#192
+        // let array_data = unsafe { array_data.build_unchecked() };
```

Comment:
How about removing this comment? Basically, there is no need to consider usages of unsafe.